### ISSUE TITLE
Aperture broken by Catalina 10.15.7 (19H1030) update

### ISSUE THREAD

**Issue by waldemarhorwat** [cormiertyshawn895/Retroactive#220]
The most recent macOS Catalina update seems to have broken Aperture's rendering of JPEG images. Old previews still appear for a split second but are replaced with gray or garbled images as Aperture re-renders the images.

**Comment by oakhampton** [comment]
Broken for me too: updated this morning and now Aperture is now showing the same symptom.
Viewing the Browser is OK, but on attempting to use 'Split View' or 'Viewer' the image appears for a very short time before disappearing leaving just the metadata. Switching off metadata (Y) shows the image again for a very short time again before it disappears, as does pressing 'M' which should show the original unedited image.

Images which have not been edited disappear to black, images which have been edited disappear to blue.

I do hope you can fix it Tyshawn.

**Comment by oakhampton** [comment]
Further to the above: Aperture can still export the Library - or parts of it like folders or projects etc as a new Library - and this can be opened in Mac Photos, from which images can be exported.

**Comment by angrenez** [comment]
The same issue for me :-(

**Comment by walterdavis** [comment]
I’m guessing this is related to the breakage of CoreImage as noted here: https://mjtsai.com/blog/2021/04/30/catalina-security-update-2021-002-breaks-opencl/

**Comment by jb59** [comment]
..and the same issue for me (iMac Catalina) - FWIW iPhoto working fine!

**Comment by AlainAvon** [comment]
That's the same issue for me !  Hope you can solve this big problem soon ...

**Comment by drbajm** [comment]
Same issue here, MacOS 10.15.7… such a pity since Retroactive  - Apple Aperture  worded so well along ! Hopefully, a solution will be provided.
Best regards from France
JMB

**Comment by drbajm** [comment]
An afterthought… wouldn't an upgrade from MacOS 10.15.7 Catalina toward MacOS Big Sure 11.31 solve the issue ? Your thoughts…
Best regards
JMB

**Comment by walterdavis** [comment]
I haven't tried that yet, but why do you suppose that would work? Were there changes to CoreImage in Big Sur that would make the bug go away again?

**Comment by drbajm** [comment]
To put it simply, in desesperation, I tried it of course without any success as you already guessed !
140.000 photos to transfer towards Lightroom… only 120.000 until the end. I already miss some cataloguing features found in the old Apple Aperture :-(
Best regards

**Comment by walterdavis** [comment]
It may not be a popular opinion, but Aperture's retouching tools were one of my most favorite things. I know Photoshop really well, I've used it since version 2 or so, but I am so productive in Aperture, and especially for studio shots, when I want to get rid of the same clump of sensor dust on 30 images, nothing even comes close. I wish that Apple had kept their eye on the Pro ball, photo wise.

**Comment by drbajm** [comment]
UPDATE — Even if the first trials using Big Sur — MacOS 11.3.1 — didn't seem conclusive at first, I was able to restore most of the files preview (delete and reconstruction). A big improvement with an apparent positive side effect (to be confirmed) : Adobe Photoshop seems no longer to be crashing !
Summary : when using my Pro computers, OS upgrades are regularly delayed until most of the applications are upgraded and the MacOS more mature. In this case it is a definite improvement but, in spite of this, I will keep switching all my image files to Adobe Lightroom.
Best regards from France
JM

**Comment by AlainAvon** [comment]
I don't know if this will help. I have 3 iMacs (two at home and one at work). All three have 10.15.7 installed. Only one has this problem with Aperture. Maybe it could be interesting to know if the problem came really with 10.15.7 ??

**Comment by drbajm** [comment]
Bonsoir
Using Aperture after upgrading from Catalina 10.15.7 toward Big Sur 11.3.1… all problems are gone ! Aperture -Retroactive seems to be fully functional even if I didn't test all features.
All in all, it worked for me even if I maintained my decision to switch entirely to Lightroom: my RAW development workflow used Lightroom since several years and I found only Aperture vastly superior as a cataloging system.
Best regards from France
JM

**Comment by gibletizer** [comment]
I have the same issue, and it started very recently, probably with the 10.15.7 **security** update, which I installed on 5/4, not the update, which I installed on 12/8/20 or the supplemental update installed on 4/2. So I suspect it's the  OpenCL issue walterdavis cites above. The page he links to lists several other issues with this update, but I don't know if Apple, in fixing those, will fix this. Until then, that page says "the quick fix is to tell Core Image not use the software renderer." How is that done? I know nothing about this stuff.  

FYI, a comment from 5/5 on https://stackoverflow.com/questions/67318465/rendering-ciimage-results-in-a-blank-image-on-macos-10-15-7-19h1030 says the OpenCL break is "fixed in macOS Catalina Security Update Developer Beta 2021-003." But this discussion lists other issues with this beta update: https://forums.macrumors.com/threads/macos-10-15-catalina-on-unsupported-macs.2183772/page-563. I could upgrade to Big Sur but I'm afraid that would pushing it on my late 2013 MBP.

**Comment by walterdavis** [comment]
The link was to a developer's blog, and the kind of change (from software to hardware rendering) is something a developer can do within their application, but not something you can adjust outside of the internals of the application. If Cormiertyshawn can fiddle that along with the thousands of other things they already fiddle, then sure, this may be relevant.

Walter

**Comment by serprinss** [comment]
is it working on 19H1030?

**Comment by walterdavis** [comment]
Yes and no. It's broken in a new way for me here on 19H1030. Instead of the large image appearing and then being replaced with "Image Format Not Supported" and a big yellow danger sign, I get a brief flash of the full-size image followed by nothing. If I resize the window, the large image appears briefly, then disappears again. Also, unlike the previous issue, the thumbnail is not affected by this. Previously, once the large image was marked "not supported", the thumbnail would also get a gray danger triangle and no longer show the image. I'm about to try upgrading to Big Sur. In case it matters at all, I'm on a 2013-series Mac Pro, which is entirely built around hardware for OpenCL. It's pretty much the poster child for that software ethos.

**Comment by walterdavis** [comment]
I decided to bite the bullet (just delivered a big project, and I don't have any deadlines ATM) and upgrade to Big Sur. I can report that after unlocking Aperture again after the upgrade, everything appears to be working w/r/t this issue. I am very relieved that I can keep on using my favorite toolchain with my pro cameras (Leaf and Nikon). Thanks to all who suggested this path, it does work.

**Comment by AlainAvon** [comment]
@walterdavis I hope upgrading to Big Sure will not be the only way. I can't do that with 3 of my Mac (too old to get the upgrade) ....

**Comment by serprinss** [comment]
> The link was to a developer's blog, and the kind of change (from software to hardware rendering) is something a developer can do within their application, but not something you can adjust outside of the internals of the application. If Cormiertyshawn can fiddle that along with the thousands of other things they already fiddle, then sure, this may be relevant. Walter
> […](#)

could this be done with a program that sits between coreimage and aperture?

**Comment by AGILBERT01** [comment]
The same issue happened to me after I updated to the recent security update. The only way I could resolve the issue was to re install Catalina OS and NOT update the latest security patch. Hopefully someone can come up with an update??

**Comment by WingsTanar** [comment]
Same issue.  However, tried walterdavis solution and did a full clean install of Big Sur from scratch.  Downloaded Aperture from the store.  Ran v1.9 of Retroactive.  No other software loaded.  Opened Aperture and although it appeared to work, when I open a folder with several hundred images in thumbnail view and then scroll down the display, it appears that only the first 100 or so are displayed - the next "block" is blank.  After a moment, the blank block gets populated with thumbnails - but then on scrolling back up the screen, the first block is now blank and does NOT get populated.  Further scrolling up and down makes no difference, only some of the blocks are displayed and no more blank blocks get populated.  The tally at the bottom of the thumbnail display reads the correct number of files loaded - so it's not a file access problem, just a display problem.

Second issue, despite my seeing somewhere that the latest version of Retroactive allows import from an external drive and allows full access to referenced files - I cannot get files on an external drive to appear available for import (and thus I cannot import, obviously).

I have seen several comments on line claiming that Aperture unboxed works fine under Big Sur, and several that it does not.  Which is it ?  The implication is that there is an underlying hardware issue - different machines responding differently.  walterdavis states he is on a 2013 Mac Pro - I am trying to work with a MacBook Pro 16 2019.  Different graphics cards ?  Or am I doing something wrong ?

**Comment by walterdavis** [comment]
Since updating to Big Sur, and re-unboxing my existing copy of Aperture, I have seen the issue with the images taking a while to load in their thumbnails, but it usually works out if I'm patient. For someone who was once so stoked with the utterly fluid display of tens of thousands of images in Aperture, it is a real let-down. At least the images no longer say "unsupported image format" any more, so I take my wins where I can. I am using a 2003-era "trash can" Pro, with two AMD FirePro D300 2GB cards in it. Those are the low-power "standard" cards for this era Mac. It doesn't seem to matter what kind of image they are. Some of the folders where I have seen it are full of JPG scans of my black-and-white film work. Others are raw Nikon D3 images. And I've also had some weird issues with my Leaf back images: sometimes they will display after import, other times they will not. When they don't, I am back to the "unsupported image format" error.

**Comment by drbajm** [comment]
Since I was the one that suggested and tried using Aperture with Big Sur , I already gave an early feedback (see earlier posts). All things measured after a much more intensive use, **Aperture is quite usable** **BUT** the blank image is also noted — one click on the "unpopulated area" corrects the problem — and Aperture isn't anymore the rock steady app I knew. Most important, the more than hundred thousand pictures can be exported either as originals or exported files (after applying the metadata since they are precious).
I own two Macs : an iMac Pro (Intel Xeon W 10 cœurs 3 GHz, 64 Go RAM, Radeon Pro Vega 64) as well as a high-end Macbook Pro 16", using the latest MacOS and Retroactive app versions… same problems on the two platforms.
As already stated, I decided to definitively switch to Lightroom… a massive amount of work — I knew why a postponed the process ! — but IMO it is worth the effort and much easier since Aperture is working again :-)
Best regards from France

**Comment by AlexMorello** [comment]
Aloha, folks, - I see the issue hasn't been resolved. Has Retroactive's developer commented on this yet?

**Comment by AlexMorello** [comment]
Hey, folks - I see the issue hasn't been resolved. Has Retroactive's developer commented on this yet? Can't instal Big Sur because my Mac isn't supported. Don't want to switch to Lightroom, either - not a fan of subscription-based software much. Also, has anyone reported this to Apple already? https://www.apple.com/feedback/macos.html

**Comment by oakhampton** [comment]
From the comments above, it seems that Big Sur is definitely not a reliable upgrade to get Aperture working again.
Fortunately I had a Carbon Copy Cloner hard drive with a Mojave clone: CCCloner reported that the Catalina upgrade changed too much to go back, so I saved the last Mojave clone to a separate drive. Since the Catalina update I've been using that as a start up disc in order to use Aperture, though it isn't a true solution.
I tried Lightroom once; never again! I actually have a copy of Lightroom 6; it is reported that it will run on Catakina, but the installer program is only 32 bit and so Lightroom can't be installed onto Catalina - but it works if it was already installed on Mojave!!

**Comment by AlexMorello** [comment]
Thought as much. We certainly need the developer to look into this. I've reported the bug to Apple, too.

**Comment by AlexMorello** [comment]
> From the comments above, it seems that Big Sur is definitely not a reliable upgrade to get Aperture working again.
> Fortunately I had a Carbon Copy Cloner hard drive with a Mojave clone: CCCloner reported that the Catalina upgrade changed too much to go back, so I saved the last Mojave clone to a separate drive. Since the Catalina update I've been using that as a start up disc in order to use Aperture, though it isn't a true solution.
> I tried Lightroom once; never again! I actually have a copy of Lightroom 6; it is reported that it will run on Catakina, but the installer program is only 32 bit and so Lightroom can't be installed onto Catalina - but it works if it was already installed on Mojave!!

Thought as much. The developer needs to look into this, for sure. Anybody got his email? I've reported the OpenCL bug to Apple, too.

Sadly, I haven't got any backups this time.

**Comment by AlexMorello** [comment]
Would [this](https://apple.stackexchange.com/questions/374846/can-mojave-be-installed-on-separate-apfs-volume-under-catalina) do as a workaround?

**Comment by WingsTanar** [comment]
Good News Update 
 
MacBook Pro 16.  Catalina. Aperture using referenced files with data on external drives.

Used Mac OS Recovery to re-install 10.15.7.  (Didn't affect data, so the day's worth of backups and archiving yesterday was wasted).

Ran Aperture - same problem except this time I got blank unsupported file images instead of totally corrupted ones.  Still couldn't import from external drive.  With the external drive disconnected everything seemed to work OK, but as soon as I connected the external drive any thumbnail that Aperture tried to render became corrupt.  So, clicking on a thumbnail to display image full screen resulted in the image corrupting.

Deleted existing version of Aperture.  Re-loaded Aperture from App Store.  Deleted back copies of Retroactive, downloaded V 1.9 and ran it.  Connected external drive and  -  it appears to work as it used to.  I can import from an external drive.  Previously corrupted images now appear as unsupported file images - and reprocessing those files WORKS.

I have made sure that my system no longer installs updates in the background.

I'll continue to exercise Aperture for the rest of the day (just because it's working now doesn't mean it still will be after lunch) and update this message stream accordingly.

**Comment by AlexMorello** [comment]
Wait - we need a solution which does not require re-installing macOS. Where has the developer got to?

**Comment by WingsTanar** [comment]
There's Good News and there's Bad News.  

Tried all afternoon and have not broken the system.  Fairly confident that all is working as it was a couple of months ago.  That's good news for me - BUT - it may be bad news for others as it does mean that there is nothing fundamentally wrong with Retroactive.   Scroll to the top of these comments and the box there states that the issue has not been assigned yet -
it works as advertised - so I do not see anyone wasting effort on fixing something that is not broken.  Apple have clearly stated that Aperture is no longer supported - so reporting the issues as a bug to Apple will just be ignored. 

My re-installation of Catalina over Catalina was completely trouble free, with no impact on installed apps or data - so for me that was a few hours well spent.

BUT - how did it work ?  My original plan was to find a pre 15.10.7 version that I could install - i.e. one that had not had the update applied which has caused this problem.  I couldn't find one so I went for the current version as available on the App Store in the vain hope that it would be a version without the faulty update which I would be asked to apply on completion of the system installation.  That hasn't happened so I assume Apple does incorporate security patches into the version available on the App Store (logical thing to do) - so the only conclusion I can draw is that "my original Catalina plus latest security patch" is NOT identical to "Apple's patched Catalina on the App Store".

**Comment by WingsTanar** [comment]
Now more bad news.  Having got the system working (see previous post) I applied latest Catalina security patch 19H1217 and that has broken Aperture again.  Different this time, though.  Go through the import from external drive routine and finally, on clicking "Import", Aperture crashes - every time.  Other functions still appear to work - and no corrupted thumbnails - but importing is a vital activity and has to work.

Hopefully this can be fixed in Retroactive.  A real shame - what other application is there out there that lets me scroll through 20,500 images in one go - and lets me display images with a selected keyword faster than I can press the enter key ?

**Comment by LGL90** [comment]
I'm having same issue as well. On MacOS 10.15.7. Any images I imported in since after the OS update show black thumbnails. Some flashed briefly when double clicked to view in full. Ugh! How do we ping or contact the Retroactive developer?

**Comment by WingsTanar** [comment]
I believe these comments are automatically e-mailed to cormiertyshawn895/Retroactive - the "developer".

**Comment by gibletizer** [comment]
Tyshawn, 

Thank you so much for creating Retroactive and please know that I for one would gladly pay at least $100 for a version that fixes this problem.

**Comment by michaelherger** [comment]
Unfortunately I believe that's the day where I have to say goodbye to Aperture. I knew that day would come, it's been inevitable. @cormiertyshawn895's work was magic, when I first found out. But there's only so much he can do. When Apple kills an API which is used at the core of an application, then he can't fix this. He was able to stub out some less central features like movie import/export. But that meant dropping a feature. Now when it comes to drawing images, that's not a feature you can drop. It's the end.

At least Photos now can handle my collection (in the early days it would chew on it for hours, before failing). But man... is this limited and buggy... My workflow, which I was able to handle all in Aperture before, now requires multiple tools (HoudahGeo, RAW Power, Better Rename). And then Photos fails to export the carefully applied keywords (see eg. https://discussions.apple.com/thread/252194855?answerId=255273768022#255273768022). They claim to have fixed it in the latest update from a few days ago. But it clearly is not fixed. Support even suggests exporting to PNG instead of JPG, because the export to PNG doesn't show that failure...

It's a sad day. But it's the day to face what we've always known: Apple stopped development of Aperture. It would break one day. Time to say goodbye.

And thank you very much @cormiertyshawn895 for the additional years with Aperture you gave us!

**Comment by AlexMorello** [comment]
> Unfortunately I believe that's the day where I have to say goodbye to Aperture. I knew that day would come, it's been inevitable. @cormiertyshawn895's work was magic, when I first found out. But there's only so much he can do. When Apple kills an API which is used at the core of an application, then he can't fix this. He was able to stub out some less central features like movie import/export. But that meant dropping a feature. Now when it comes to drawing images, that's not a feature you can drop. It's the end.
> 
> At least Photos now can handle my collection (in the early days it would chew on it for hours, before failing). But man... is this limited and buggy... My workflow, which I was able to handle all in Aperture before, now requires multiple tools (HoudahGeo, RAW Power, Better Rename). And then Photos fails to export the carefully applied keywords (see eg. https://discussions.apple.com/thread/252194855?answerId=255273768022#255273768022). They claim to have fixed it in the latest update from a few days ago. But it clearly is not fixed. Support even suggests exporting to PNG instead of JPG, because the export to PNG doesn't show that failure...
> 
> It's a sad day. But it's the day to face what we've always known: Apple stopped development of Aperture. It would break one day. Time to say goodbye.
> 
> And thank you very much @cormiertyshawn895 for the additional years with Aperture you gave us!

Wait - did Apple actually kill something this time? I thought it was a mere bug in the latest security update. Retroactive still works on Big Sur, doesn't it? Anyway, it's high time @cormiertyshawn895 commented on this, at last! 

P.S. One could also do a clean Catalina installation to fix this, couldn't they?

**Comment by michaelherger** [comment]
> Wait - did Apple actually kill something this time? I thought it was a mere bug in the latest security update.

Whether they introduced a bug or killed it on purpose doesn't really change much IMHO: they have announced the end of Aperture, _and_ they have announced the end of some features that Aperture seems to be using in its core.

**Comment by AlexMorello** [comment]
> > Wait - did Apple actually kill something this time? I thought it was a mere bug in the latest security update.
> 
> Whether they introduced a bug or killed it on purpose doesn't really change much IMHO: they have announced the end of Aperture, _and_ they have announced the end of some features that Aperture seems to be using in its core.

Have they? If it is a bug, they will fix it in future updates. Security patches have nothing to do with OpenCL. Also, they haven't announced anything regarding Aperture. It was abandoned back in 2015 or so. 

Upd: it appears they did it on purpose:

'Both updates address memory corruption and handling issues with the kernel that could elevate privileges or allow kernel memory disclosure, a Wi-Fi-related memory corruption issue that could allow an app to cause unexpected system termination or write kernel memory, and a Windows Server issue that could leak a user’s credentials from secure text fields.'

**Comment by AlexMorello** [comment]
@cormiertyshawn895 Please comment on this ASAP - the users may re-instal the OS if it is intentional ☝🏻

**Comment by AlexMorello** [comment]
Has anyone tried installing the new 2021-003 security update?

**Comment by AlexMorello** [comment]
> Unfortunately I believe that's the day where I have to say goodbye to Aperture. I knew that day would come, it's been inevitable. @cormiertyshawn895's work was magic, when I first found out. But there's only so much he can do. When Apple kills an API which is used at the core of an application, then he can't fix this. He was able to stub out some less central features like movie import/export. But that meant dropping a feature. Now when it comes to drawing images, that's not a feature you can drop. It's the end.
> 
> At least Photos now can handle my collection (in the early days it would chew on it for hours, before failing). But man... is this limited and buggy... My workflow, which I was able to handle all in Aperture before, now requires multiple tools (HoudahGeo, RAW Power, Better Rename). And then Photos fails to export the carefully applied keywords (see eg. https://discussions.apple.com/thread/252194855?answerId=255273768022#255273768022). They claim to have fixed it in the latest update from a few days ago. But it clearly is not fixed. Support even suggests exporting to PNG instead of JPG, because the export to PNG doesn't show that failure...
> 
> It's a sad day. But it's the day to face what we've always known: Apple stopped development of Aperture. It would break one day. Time to say goodbye.
> 
> And thank you very much @cormiertyshawn895 for the additional years with Aperture you gave us!

Mate, can't you just re-instal macOS and steer clear of those 'security' updates from now on? I've just confirmed the App Store version doesn't contain them - at least with Catalina.

**Comment by WingsTanar** [comment]
Tried 2021-003 otherwise known as 19H1217 three days ago.  Aperture crashes on trying to import.  See my comment 3 days ago in this chain.

Re-installing Catalina from AppStore DOES result in a working solution.  i.e. revert to the point before the 19H1030 security update broke Aperture.  Having re-installed (using OS Recovery so not affecting existing data or applications) delete all the Aperture installation (not the data file, of course) - re-load Aperture from App Store and re-apply Retroactive 1.9.

**Comment by AlexMorello** [comment]
Seems the issue is not solely with Apple this time. These here read OpenCL has been fixed in 2021-003:

https://stackoverflow.com/questions/67318465/rendering-ciimage-results-in-a-blank-image-on-macos-10-15-7-19h1030

https://shapeof.com/archives/2021/4/apple_april_2021_security_update_and_acorn.html

@cormiertyshawn895, please show some respect for the community and comment on this at long last!

**Comment by michaelherger** [comment]
@AlexMorello please show some respect for people who helped you delay the announced end of your beloved software application. For free.

**Comment by AlexMorello** [comment]
@cormiertyshawn895 could at least drop a line or two in here

**Comment by rologoetz** [comment]
I`m still using Aperture frequently on my iMac 27" (late 2013) which is using Catalina 10.15.7 and I noticed the same behavior as others mentioned above after some automatic "safety updates" from Apple :-(

Reading the posts above I just want to be sure : do I really have to completely do a new reinstall with Catalina again (including all other stuff) to get back to the "original" Catalina to get Aperture back again ? 

Or is there another way to only reinstall the system without reinstalling all the rest again ? 

And I agree to @AlexMorello : any comment by @cormiertyshawn895  would be very nice - at least to see if there is a chance that this issue gets solved or whether  this is the real final end for Aperture on my Mac :-(

**Comment by WingsTanar** [comment]
Yes there is a way of only installing the OS. 

I first uninstalled the Aperture app (having ensured that I could reinstall it from my App Store account) and made a backup of the Aperture library, then used Mac Recovery on my MacBook Pro 16 to re-install 10.15.7 from the App Store just as if I were applying an upgrade.  None of the installed applications or data were affected.  I then reinstalled Aperture and ran a fresh copy of Retroactive set to the Catalina option.

Of course, I might just have been lucky to catch the version of 10.15.7 without the safety update installed or (as I postulated in a previous post in this chain) a clean install of the AppStore version of 10.15.7 is somehow different from the previously installed-then-updated version - but either way I am now running Aperture just as I did several months ago.

I also made a bootable copy of the installation which was a good plan because when the next safety update was issued a week or so ago, I tried applying it to my system only to find that it also broke Aperture (but in a different way).  Having the bootable backup made reversion a lot easier.

**Comment by rologoetz** [comment]
@WingsTanar thx a lot for the details ! 

I followed your instructions and reinstalled Catalina via recovery partition (restart with cmd-R). Worked fine so far. After rebooting MacOS I immediatelly removed the tick at "automatic updates" in the system settings. 
Then I deleted the old Aperture app (put into trash). Next step was to get Aperture 3.6 by new which wasn't that easy, because it doesn't appear anymore in my AppStore account despite I've bought it years ago :-( 
As I couldn't find another source in the net for Aperture 3.6 I finally recalled that I've a copy of it on my old 27" iMac (mid 2010 with 10.13), so I simply dragged this on an USB stick and copied it to my newer iMac. After this I restarted Retroactive to do the patch. 
Et voilà : after restarting Aperture the pictures show again as they did before. Some previews have been obviously corrupted by the system troubles (they only showed some random noise), but I could create the previews by new by simply marking them, going to context menu + simulantiously pressing the option (alt) button + selecting "create preview". 

So everything is fine again - thank you for your input ! 

Nevertheless system is now reminding me again to the safety update ... so I'd be very happy if someone finds a solution to do this without corrupting Aperture again :-(

**Comment by cpdesjardins** [comment]
> Unfortunately I believe that's the day where I have to say goodbye to Aperture. I knew that day would come, it's been inevitable. @cormiertyshawn895's work was magic, when I first found out. But there's only so much he can do. When Apple kills an API which is used at the core of an application, then he can't fix this. He was able to stub out some less central features like movie import/export. But that meant dropping a feature. Now when it comes to drawing images, that's not a feature you can drop. It's the end.
> 
> At least Photos now can handle my collection (in the early days it would chew on it for hours, before failing). But man... is this limited and buggy... My workflow, which I was able to handle all in Aperture before, now requires multiple tools (HoudahGeo, RAW Power, Better Rename). And then Photos fails to export the carefully applied keywords (see eg. https://discussions.apple.com/thread/252194855?answerId=255273768022#255273768022). They claim to have fixed it in the latest update from a few days ago. But it clearly is not fixed. Support even suggests exporting to PNG instead of JPG, because the export to PNG doesn't show that failure...
> 
> It's a sad day. But it's the day to face what we've always known: Apple stopped development of Aperture. It would break one day. Time to say goodbye.
> 
> And thank you very much @cormiertyshawn895 for the additional years with Aperture you gave us!

I followed your advice and launched a trial of Capture One.  It imported my entire Aperture Library flawlessly and preserved all the Project and Album structure.  Brilliant!

I think Capture One will be my future and I too am thankful to @cormiertyshawn895 for the additional years with Aperture he gave us.  Long enough to gracefully depart with no regrets.

**Comment by waldemarhorwat** [comment]
I can confirm that Catalina security update 3 «10.15.7 (19H1217)» fixes the Aperture rendering problem introduced by Catalina security update 2 «10.15.7 (19H1030)». Aperture now works as it did prior to that security update and once again I can view and render images.

I can import images just fine into Aperture in Catalina 10.15.7 (19H1217). There is an unrelated years-old problem that causes crashes if I try to confirm the import dialog while Aperture is working on other things (including generating the thumbnails for the import dialog), but it does work for me as long as there is no other Aperture activity when I confirm the import dialog.

**Comment by AlexMorello** [comment]
@waldemarhorwat What have you done besides updating, then? Pictures still disappear after 'loading' for me.

**Comment by AlexMorello** [comment]
> Capture One.

$349? LOL

**Comment by AlainAvon** [comment]
@waldemarhorwat    Same question as @AlexMorello . New security update doesn't fix the Aperture rendering problem for me.

**Comment by rologoetz** [comment]
@waldemarhorwat @AlexMorello just installed the security update 2021-003 that MacOS was proposing (guess that's what you mean if you write secure update 3 «10.15.7 (19H1217)» ???) after I did the new clean install of Catalina (as mentioned Aperture then worked again). But : again the same mistake in Aperture :-(

So I've now to reinstall Catalina & Aperture again ...

**Comment by walterdavis** [comment]
There are free options available for individual camera/sensor versions. I have a Leaf back I purchased third-hand, and I am using a completely free version of C1 with it. Support (once you figure out the incredible chain of ownership around Leaf the company) is excellent, even getting me a download of an installer for a mid-version update that allowed me to use FireWire tethering despite the official download of version 12 removing that feature. They are currently on version 21 or something, but my vintage gear doesn't play well with the latest.

Walter

**Comment by AlexMorello** [comment]
Those special versions only support RAW. There are cracked versions of Capture One all over the Internet, though :D 

Also, I really want to call @cormiertyshawn895 names now. With all due respect, he could at least have chimed in on this, seeing as he's been tagged here like a dozen times.

**Comment by WingsTanar** [comment]
There may be a thousand perfectly good and understandable reasons for Shawn not to have joined in on this discussion - not least that, as several people have now confirmed, Retroactive does indeed make Aperture usable under Catalina.  So, there is nothing identifiably 'wrong' that Shawn could 'fix'. 

On the other hand, I can understand the frustration that some feel.  I stuck with my 2009 Mac Book running El Capitan long after I would have liked to upgrade simply because I wanted to run Aperture.  I only jumped in and got a Mac Book Pro 16 after I saw items on the web 'advertising' Retroactive.  To then find Aperture collapsing less than a year later was soul destroying.  I was also dismayed to find that even though it worked (initially) under Catalina it does not work for me under a completely clean install of Big Sur with fresh copies of Aperture and Retroactive.  C'est la vie.
 
Anyway, those of you considering giving up on Aperture, you might like to look at the 30 day free trial of Photo Supreme.  As far as I can determine it is possible to configure this Digital Asset Management program to emulate the cataloguing and searching functions of Aperture very closely.  I can't comment on the (fairly new, I am told) picture editing functions - but then I never used them in Aperture - preferring to use a dedicated program for editing (Photoshop, Affinity etc) and keeping Aperture as a dedicated DAM program.  There would appear to be a fairly steep learning curve - indeed one might be forced to do the unmanly thing and read the manual - and I have only tried rebuilding a database from scratch so far.  Once I have the hang of the interface I might try a full import from Aperture - but that's for another day.

Bye y'all.

**Comment by AlexMorello** [comment]
Bingo! I can confirm doing a clean instal - that is, wiping the drive, re-installing Catalina, and EVEN installing the 003 security update - does the trick. After downloading from the App Store and applying Retroactive, Aperture is working again. 

So:

- Back your files up
- Wipe the drive: in Recovery Mode, launch Disk Utility and **delete** the volume 'Macintosh HD - Data'. Restart the Mac, boot into Recovery again, and **erase** Macintosh HD.  
- Close Disk Utility and proceed with installing Catalina. I didn't use Internet Recovery, but rather downloaded the App Store version (10.5.7).
- Apply the 003 security update 
- Download Aperture from your purchase history in the App Store
- Apply Retroactive 1.9 
- If all is well, import your files back in.  

Thanks everyone for your input!

**Comment by waldemarhorwat** [comment]
> @waldemarhorwat What have you done besides updating, then? Pictures still disappear after 'loading' for me.

I don't know how much of this matters, but here's exactly what I did:

After discovering Aperture failing to render anything in Catalina 10.15.7 (19H1030), I reinstalled Catalina from my recovery partition, which happens to revert Catalina back to 10.15.7 (19H15) on my Mac. I then manually updated it only as far as Catalina 10.15.7 (19H512) so I could still use Aperture, and started this GitHub thread. Things stayed that way for over a month.

When Security Update 3 came out, I just let Catalina 10.15.7 (19H512) update itself to the latest version via Software Update. Thus I got to Catalina 10.15.7 (19H1217) and Aperture still renders correctly.

**Comment by gibletizer** [comment]
I installed the 003 security update and Aperture seems to be working perfectly fine again. The only thing I had to do was to re-import a couple images I had imported after the 002 update that had corrupted preview thumbnails. I did nothing else. Halleluyah!

**Comment by AGILBERT01** [comment]
I too just installed 003 security update and was able to open view and use Aperture as before! SOOOOOO releived and happy that I have a new lease on using Aperture however brief it still may be.

**Comment by visorz** [comment]
Still broken preview images and black single image view with **macOS 10.15.7 (19H1323)**. I used my installed version of Aperture 3.6 with **Retroactive 1.9**.

**Comment by gibletizer** [comment]
Does anyone know if 005 affects Aperture?

**Comment by WingsTanar** [comment]
005 installed automatically yesterday. Aperture appears unaffected.

**Comment by visorz** [comment]
I had to reinstall Catalina from network recovery (10.15.7 19H15) and then disable further security updates. Before that I tried security update 006 (19H1419) which broke Aperture rendering. I am unable to upgrade to BigSur because my MacBook Air is from mid 2012. 

Is it technically possible to relink the broken rendering library to a self coded one?

**Comment by visorz** [comment]
@WingsTanar: Did you do anything special to have a working Aperture after security update 005? What system do you use?

**Comment by WingsTanar** [comment]
@visorz;  Nothing at all, 005 just worked.

MacBook Pro 16.  Catalina 10.15.7.  Current update 005 (006 has just been offered but not yet installed.)
Every attempt I have made to install Aperture under Big Sur has failed in some way or another - usually in failing to display thumbnails.

**Comment by WingsTanar** [comment]
Now installed update 006 - Aperture still functioning correctly

**Comment by gibletizer** [comment]
just installed 006, Aperture seems fine. Whew!

**Comment by theirix** [comment]
Experienced this bug on 10.15.7 with 007 installed (19H1519). Tried to redownload Aperture and reapply a Retroactive patch - it did not help.

**Comment by gibletizer** [comment]
I put 007 on 10.15.7 yesterday so I’m also running 19H1519 and Aperture seems fine.

**Comment by WingsTanar** [comment]
Just installed 007 (19H1519) on 10.15.7 and Aperture appears to be OK.

theirix, you might try my solution from 3 Jun 

**_I first uninstalled the Aperture app (having ensured that I could reinstall it from my App Store account) and made a backup of the Aperture library, then used Mac Recovery on my MacBook Pro 16 to re-install 10.15.7 from the App Store just as if I were applying an upgrade. None of the installed applications or data were affected. I then reinstalled Aperture and ran a fresh copy of Retroactive set to the Catalina option_**

The essence is that you install Aperture and run Retroactive against a CLEAN install of Catalina. Redownloading Aperture over a previous installation of Aperture (even though it was uninstalled) seems not to work.

**Comment by theirix** [comment]
@WingsTanar thank you! I was curious if a simple Aperture reinstall could help - it cannot, only a clean reinstall of OS.

**Comment by cormiertyshawn895** [comment]
This has been fixed in [Security Update 2022-005 Catalina](https://support.apple.com/kb/HT213343). 

If you are experiencing this issue, open System Preferences > Software Update > More Information..., and install the latest macOS Catalina security update.

**Comment by gibletizer** [comment]
Thank you so much!!!

**Comment by AlainAvon** [comment]
> This has been fixed in [Security Update 2022-005 Catalina](https://support.apple.com/kb/HT213343).
> 
> If you are experiencing this issue, open System Preferences > Software Update > More Information..., and install the latest macOS Catalina security update.

Unfortunately this doesn't work !   Exactly the same issue as before with Aperture's rendering of JPEG images...

**Comment by Grizzletoast** [comment]
I am also still having this issue, 10.15.7 19H2026 unboxed with Retroactive 2.0 (first used 1.9, then unlocked again with 2.0), there are no updates available in Software Update. [Security Update 2022-005 Catalina](https://support.apple.com/kb/HT213343) is confirmed as installed according to System Information > Software > Installations.

**Comment by WingsTanar** [comment]
I cannot prove it, but I am convinced that there are two "versions" of 10.15.7.  There is the version you get when you apply updates to previous versions and there is the version you get when you use Mac Recovery to re-install the whole operating system.  At some time in the past (when all these Aperture issues first arose) a faulty update corrupted code in an area that the update should not have been affecting. As this corruption occurred in code that is NOT normally changed by an update, subsequent updates have had no effect on correcting the issue.

The solution I used was to re-install ALL of the operating system (Look back at my comment of 30 Nov 2021), not just to rely on subsequent updates to correct the problem.

Happy New Year

**Comment by georgrehm** [comment]
Is there any fix for this? I've just updated an old (2013) iMac to Catalina and unlocked Aperture. Viewing photos doesn't work and exporting photos doesn't work either. In short, Aperture is completely broken right now. If there is a fix I'd be very happy about a pointer.

**Comment by AlexMorello** [comment]
> Is there any fix for this? I've just updated an old (2013) iMac to Catalina and unlocked Aperture. Viewing photos doesn't work and exporting photos doesn't work either. In short, Aperture is completely broken right now. If there is a fix I'd be very happy about a pointer.

Honestly, I wouldn’t bother. I’d been holding out for a very long while, and it wasn’t until I had finally migrated to Luminar 4 that I realised how much progress the algorithms had made in the meantime. Luminar is by no means perfect, but its ‘AI’ often does in an instant what would require numerous manual adjustments with Aperture. I guess it IS time to move on at last.

P.S. The 2013 iMac isn’t really old ;)

**Comment by WingsTanar** [comment]
How have you updated to Catalina ?  If you have incrementally updated (applying update after update) then you may still be stuck with a corrupted operating system.  I forget which particular update caused the problem, but it seems that update changed something it should not have changed - in an area that doesn’t get corrected in subsequent updates. (Read back the history on GitHub.)

Try a complete system install using the version available on the net - then reinstall a clean version of Aperture - then apply retroactive.

I'm running Aperture under Catalina 10.15.7 with (touch wood) no problems so there is nothing wrong with Aperture itself.  Others who have gone for the complete re-install route also reported success.

Keep trying - Aperture is still the best digital asset management program out there.  If you want an image processing package, then  I’ll admit that Aperture is not the one to go for.

Good luck.

**Comment by audio-inside** [comment]
I am also having the same corrupt previews and thumbnails issue even after having done a removal of Aperture (by dragging to the Trash) a restore of Catalina 10.15.7 with an application of the latest Security Update 2022-005, then a "reinstall" of Aperture. 

I accomplished the "reinstall" by dragging the app over from a machine where it is running correctly on Mojave, but I see some instructions here suggesting better results with a FRESH install of Aperture before applying Retroactive to it. However, the Installer refuses to go ahead with the install citing the incompatibility of Aperture with the current operating system.  

How did people work around this?

**Comment by WingsTanar** [comment]
For me, Aperture is still available for download from my App Store account.

**Comment by audio-inside** [comment]
> For me, Aperture is still available for download from my App Store account.

Yes I already have multiple archived copies of the installer.  I just don't know how to PERFORM the installation when I'm already running Catalina - the installer refuses to go forward after detecting that I'm running a MacOS version that's incompatible with Aperture.

**Comment by WingsTanar** [comment]
Interesting.   I don't have an "installer" - by which I assume you mean a dmg file.  I merely click on the download cloud next to the Aperture entry in my App Store account.  If you purchased Aperture from the App Store then it should still be available to download/install - unless you purchased it in a box with a physical DVD from an Apple shop.

Anyway, you  say you restored 10.15.7 and then applied an update.  But if you restored Catalina from the version that is called by the Mac OS Recovery option then it should have been a fully updated copy.  That makes me wonder if you did indeed restore using the Mac OS Recovery option (Restart your Mac and hold down the Command and R keys to boot into Recovery Mode.).  

If you restored using any other mechanism then you cannot guarantee the version you are restoring from does not have the remains of the faulty update that caused all these problems in the first place.

I know it's not much help to be told "works for me" - but that's the truth of it.  I did a Catalina Mac OS Recovery which retained all my data and apps - downloaded Aperture from the App Store - ran retroactive 1.9 (with the right options set) and have Aperture 3.6 running fine on my MacBook Pro.

However - took a while to import all my images again which reminds me that all my images were referenced.  If your images are stored in your Aperture library make sure you have a couple of copies of your library backed up externally to your Mac - you should be able to reconnect the library to Aperture (if you can get Aperture to install - good luck).

**Comment by audio-inside** [comment]
> Anyway, you say you restored 10.15.7 and then applied an update. But if you restored Catalina from the version that is called by the Mac OS Recovery option then it should have been a fully updated copy. That makes me wonder if you did indeed restore using the Mac OS Recovery option (Restart your Mac and hold down the Command and R keys to boot into Recovery Mode.).

Thanks for these details.  Yes I restored 10.15.7 from the Recovery partition (restart, CMD-R).  That's a process I'm not too savvy about - previously I've only done individual files or complete machine restores from Time Machine backups - so I didn't know exactly which version of 10.15.7 I was getting.  After I did the restore it appeared that the Security Update 2022-005 had NOT been applied so I ran that on top of the restore.  Then did the App Store "install" of Aperture 3.6 via the "cloud download" and got the Open button - this was also a new procedure for me - but Open seemed to produce no activity, and obviously Aperture did not open.  I have not yet checked Console logs to see what the nature of the error might be.

**Comment by WingsTanar** [comment]
You mention "restore`, 'update' and download - but do not mention applying Retroactive.  It is the running of Retroactive that modifies the Aperture installation such that it will run under Catalina.

'open' might not work with an unmodified Aperture.  However, is there an "Aperture App" in your applications folder ?   If there is then you should be able to download and run Retroactive.

**Comment by audio-inside** [comment]
> You mention "restore`, 'update' and download - but do not mention applying Retroactive. It is the running of Retroactive that modifies the Aperture installation such that it will run under Catalina.
> 
> 'open' might not work with an unmodified Aperture. However, is there an "Aperture App" in your applications folder ? If there is then you should be able to download and run Retroactive.

Thanks again for your help, I started again "from scratch" as best I could by:
1) Deleting the existing Aperture.app
2) Restoring Catalina 10.15.7
3) Applying Security Update 2022-005 
4) Downloading (but NOT opening) Aperture from the App Store
5) Applying Retroactive 2.0 to Aperture then quitting Retroactive.
6) Opening Aperture.app from the Applications folder
7) Importing a folder of JPG files that were already on my desktop (not external drive) with "New Projects Automatically Generate Previews" enabled in Preferences at a relatively small size 1280x1280

Unfortunately I get substantially the same results, with MOST thumbnails successfully created and the full-size Preview image disappearing after a few seconds.  Note there was an interesting phenomenon that thumbnails of any image taken in portrait orientation was corrupted - in addition, any Previews or thumbnails re-generated for any image AFTER the import process are also corrupted.
![image](https://user-images.githubusercontent.com/9272025/214902381-30e3cbe5-24c2-4e77-b39f-ab367a73c341.jpeg)

I wonder if there is an underlying issue with CoreImage and OpenCL on my specific hardware?  I'm running a MacBook10,1 which includes a Metal-supported Intel HD Graphics 615 engine with feature set macOS GPUFamily2 v1 on a 2304 x 1440 Retina display. 
![image](https://user-images.githubusercontent.com/9272025/214902134-7a93c313-d95a-4001-9a0c-e64c2b304f43.jpeg)

**Comment by WingsTanar** [comment]
Ahaa - you are now in the position I was in when the whole problem started.  One day all was working fine, the next the thumbnails were corrupted.  One of the early entries in this chain on GitHub said

" I’m guessing this is related to the breakage of CoreImage as noted here: https://mjtsai.com/blog/2021/04/30/catalina-security-update-2021-002-breaks-opencl/ "

That comment led to the idea of reverting Catalina to prior to the 2021-002 update - which worked.  Personally, I never re-applied the 2021-002 update but I did eventually accept a later security update which did not break Aperture.

This also led to the idea that the bit image ending up in the laptop from working sequentially through security updates (i.e. normal progression) was NOT exactly the same as the bit image obtained by re-installing Catalina using the system recovery mechanism - perhaps because update 2021-002 impacted adversely on an area of code that it shouldn't have touched and which subsequent updates did not correct.

Unfortunately, the process you have just gone through was the process I used to fix the issue - which leaves me without any further advice - EXCEPT it might be worth trying to do it all again but WITHOUT applying 2022-005. 

Sorry

**Comment by RafaSP** [comment]
I have the same issue on Catalia 19H2026.
If I run Apperture from the terminal I get this errors, I still don't know what they mean. buy everytime I click on a image I got a new error message (the same)

objc[6285]: Class ISBundle is implemented in both /System/Library/PrivateFrameworks/IconServices.framework/Versions/A/IconServices (0x7fff91946fb0) and /Applications/Aperture.app/Contents/Frameworks/PrintServices.framework/Versions/A/PrintServices (0x10f82c3d8). One of the two will be used. Which one is undefined.
2023-01-28 10:55:10.550 Aperture[6285:315088] asked to move command sets to sandbox location, but Aperture is unboxed, skipping it
2023-01-28 10:55:11.949 Aperture[6285:315088] It's not legal to call -layoutSubtreeIfNeeded on a view which is already being laid out.  If you are implementing the view's -layout method, you can call -[super layout] instead. Break on void _NSDetectedLayoutRecursion(void) to debug.  This will be logged only once.  This may break in the future.
2023-01-28 10:55:12.522 Aperture[6285:315088] kick delayedFinishLaunching to the next run loop, so that the adjustments menu can populate
** OpenCL Error Notification: [CL_DEVICE_NOT_AVAILABLE] : OpenCL Error : Error: build program driver returned (-2) **
** OpenCL Error Notification: OpenCL Warning : clBuildProgram failed: could not build program for 0xffffffff (Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz) (err:-2) **
** OpenCL Error Notification: [CL_BUILD_ERROR] : OpenCL Build Error : Compiler build log:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.
 **
Native OpenCL program build failed:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.** OpenCL Error Notification: [CL_DEVICE_NOT_AVAILABLE] : OpenCL Error : Error: build program driver returned (-2) **
** OpenCL Error Notification: OpenCL Warning : clBuildProgram failed: could not build program for 0xffffffff (Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz) (err:-2) **
** OpenCL Error Notification: [CL_BUILD_ERROR] : OpenCL Build Error : Compiler build log:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.
 **
Native OpenCL program build failed:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.** OpenCL Error Notification: [CL_DEVICE_NOT_AVAILABLE] : OpenCL Error : Error: build program driver returned (-2) **
** OpenCL Error Notification: OpenCL Warning : clBuildProgram failed: could not build program for 0xffffffff (Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz) (err:-2) **
** OpenCL Error Notification: [CL_BUILD_ERROR] : OpenCL Build Error : Compiler build log:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.
 **
Native OpenCL program build failed:
CVMS_ERROR_INVALID_CONNECTION: Invalid connection given.

**Comment by WingsTanar** [comment]
As suggested elsewhere, this could be an update error.  Go to
https://www.lifewire.com/use-os-x-combo-updates-to-correct-installation-problems-2260836
for an explanation.  Link to latest combo update is
https://support.apple.com/kb/DL2052?locale=en_GB
Earlier combo updates are available from Apple as well.  (Combo updates refresh the operating system without affecting data - but you might like to backup your images and your Aperture library before updating).

Ensure that Aperture is available to you for download from your account on the App Store - then remove Aperture from your system.  Download the combo update (takes forever).  Fresh install Aperture, apply Retroactive.
 
Good luck

**Comment by RafaSP** [comment]
I did this:
Upgrade from mojave to catalina 10.15.7 19H15 - Aperture OK
Security update 2021-001 19H524 - Aperture OK
Security update 2021-005 - 19H1417 - OpenCL error
Security update 2022-005- 19H2026 - OpenCL error

I'm going back to 19H524, that  leaves me with a security update lower than with Mojave, but I need catalina to install a VPN

**Comment by WingsTanar** [comment]
Getting there.  If you have the energy, try it all again but miss out 19H1417 and try a later update.  As far as know you do not have to install EVERY update.  Obviously it is 19H1417 that is causing the problem.

I know that when I went through all of this, I missed out the update that I thought was causing the issue - everything still worked, and I have applied security updates since then with no problem.

Cheers.

**Comment by RafaSP** [comment]
I don't think that going directly to Security update 2022-005- 19H2026 is going to solve it, because the first time I encountered the problem was after an upgrade from mojave and then an update from the "system update" control panel that installed the last 2022-005 security update. I don't know if this OpenCL is hardware dependent I'm on a mid 2012 15" macbook pro (non retina)

Here https://codeobsession.com/news/2021/04/28/catalina-security-update-2021-002/ at the end mentions that systems with an Nvidia GPU can't go past 19H524 because later updates won't solve the problem

**Comment by Tord555** [comment]
Use Big Sur, no problems as yet with Aperture (with Retroactive, of course)!

**Comment by minaindrek** [comment]
When I use Aperture in Big Sur, first it seems there are no problems, it opens all raw files, previews come up with a little delay, but they come eventually, but when you put on a little pressure - shuffle a lot of images quickly, do some filter dragging or something similar - it tends to crash. Same thing in Catalina. Unfortunately the most stable Aperture, as it seems, is still in Mojave.
And the loupe thingy - it has frame missing... But i still love to use it.

**Comment by audio-inside** [comment]
Using Retroactive on my MacBook Air (MacBook10,1) I am getting many crashes from deep within the Apple Intel Graphics driver stack - not immediately or repeatably, but as @minaindrek mentions if I "put on a little pressure" eg while brushing in Retouch at 100% Zoom I get an Illegal instruction or other error.  See attached crash log generated during horizontal scrolling in Split View mode.  (https://github.com/cormiertyshawn895/Retroactive/files/11467512/Crashlog.txt)

Also, Aperture/Retroactive is not recognizing .MP4 video files, deeming them "unsupported" when I try and import them.
